iOS/Flows/Conversation/Cells/MessagesTimeMachineCollectionViewLayout.swift

```diff
@@ -0,0 +1,185 @@
+            // Dark the item as it moves away
+            backgroundBrightness = lerp(abs(normalizedZOffset),
+                                        start: self.frontmostBrightness,
+                                        end: self.backmostBrightness)
+        } else {
+            // Items should be at full when at the front of the stack.
+            backgroundBrightness = self.frontmostBrightness
+        }
+
+        let detailAlpha = 1 - abs(normalizedZOffset) / 0.2
+        let textViewAlpha = 1 - abs(normalizedZOffset) / 0.8
+
+        // The most recent visible item should be white.
+        if let itemFocusPosition = self.itemFocusPositions[indexPath] {
+            let normalizedFocusDistance = abs(itemFocusPosition - self.zPosition)/self.itemHeight.half
+
+            backgroundBrightness += lerpClamped(normalizedFocusDistance,
+                                                start: 1-self.frontmostBrightness,
+                                                end: 0)
+        }
+
+        // If there is no message to display for this index path, don't show the cell.
+        if self.dataSource?.getTimeMachineItem(forItemAt: indexPath) == nil {
+            attributes.alpha = 0
+        }
+
+        attributes.backgroundColor = .white
+        attributes.brightness = backgroundBrightness
+        attributes.shouldShowTail = indexPath.section == 0
+        attributes.bubbleTailOrientation = indexPath.section == 0 ? .up : .down
+        attributes.detailAlpha = detailAlpha
+        attributes.messageContentAlpha = self.isShowingDropZone && indexPath.section == 1 ? 0.0 : textViewAlpha
+
+        return attributes
+    }
+
+    // MARK: - Attribute Helpers
+
+    func getDropZoneColor() -> Color? {
+        guard let ip = self.getFrontmostIndexPath(in: 1),
+                let attributes = self.layoutAttributesForItem(at: ip) as? ConversationMessageCellLayoutAttributes else {
+                    return nil
+                }
+
+        return attributes.backgroundColor == .white ? .darkGray : .white
+    }
+
+    func getBottomFrontMostCell() -> MessageSubcell? {
+        guard let ip = self.getFrontmostIndexPath(in: 1), let cell = self.collectionView?.cellForItem(at: ip) as? MessageSubcell else { return nil
+        }
+        return cell
+    }
+
+    func getDropZoneFrame() -> CGRect {
+        let center = self.getCenterPoint(for: 1, withYOffset: 0, scale: 1)
+        let padding = Theme.ContentOffset.short.value.doubled
+        var frame = CGRect(x: padding.half,
+                           y: 0,
+                           width: self.collectionView!.width - padding,
+                           height: MessageContentView.bubbleHeight - padding)
+        frame.centerY = center.y - padding - Theme.ContentOffset.short.value
+        return frame
+    }
+
+    // MARK: - Content Offset Handling/Custom Animations
+
+    /// If true, scroll to the most recent item after performing collection view updates.
```

Comment:
The rest is just translated directly from the base class. All this scrolling and animation logic is too conversation specific to be in the base class.